lib/registry-sync/ecr-mirror.ts

```diff
@@ -187,14 +185,8 @@ export class EcrRegistrySync extends Construct {
 
   }
 
-  /**
-   * Grant the specified grantees pull privileges to the target ECR repositories.
-   */
-  public grantPull(...grantees: iam.IGrantable[]) {
-    for (const grantee of grantees) {
-      this._grantAuthorize(grantee);
-      this._repos.forEach(p => p.grantPull(grantee));
-    }
+  public ecrRepository(source: MirrorSource): ecr.IRepository | undefined {
+    return this._repos.get(source);
```

Comment:
Doesn't this mean:

```ts

const superchain1 = MirrorSource.fromDockerHub('jsii/superchain');
const superchain2 = MirrorSource.fromDockerHub('jsii/superchain');

const mirror = new EcrMirror(this, 'Mirror', {
  ...,
  images: [superchain1]
})

const repo1 = mirror.ecrRepository(superchain1); // works
const repo2 = mirror.ecrRepository(superchain2); // fails 
```

This is unexpected behavior IMO and can lead to some quirky bugs.

I think is better to implement a `toString()` on the `MirrorSource` class and have `this._repos` be a `Map<string, ecr.IRepository>`.

Reply:
Fair concern. I have the same one. I'll improve.